ubccr/xdmod
# Remove `UserRoles` Usage

## Description
**NOTE: This PR is based on https://github.com/ubccr/xdmod/pull/760 and should not be merged until #760 is.**

Just removing and or replacing `UserRoles` usage in XDMoD. This includes changes in `XDStatistics` and `XDUser`. 

The updates in `XDUser` are just removing table maintenance code. 

The changes in `XDStatistics` include updating the SQL to utilize `user_acls` vs. `UserRoles` and also to do a bunch of the work that was previously being done in PHP. 

Also removed the etl configuration file responsible for managing `UserRoles`.

## Motivation and Context
`UserRoles` is no longer used so we're removing code that references it. 

## Tests performed



All automated tests w/ fresh install && upgrade
- unit
- component
- integration
- regression

Manual testing was done for XSEDE install
- Navigate to the Internal Dashboard
- Select `User Management`
- Select `User Stats`
- Select `Internal Users` user type
- Ensure that data is displayed as expected / correctly.
- Log in with a few users
- Navigate back to `User Stats`
- Ensure that the data has been updated appropriately. 

## Types of changes

- [X] Code Cleanup / Refactor

## Checklist:
